bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -118,42 +128,24 @@
     private Map<String, Set<String>> mapModuleTypeToRules = new HashMap<String, Set<String>>();
 
     /**
-     * {@link Map} of created rules. It contains all rules added to rule engine independent if they are initialized or
-     * not. The relation is rule's id to {@link Rule} object.
-     */
-    private Map<String, RuntimeRule> rules;
-
-    /**
      * {@link Map} system module type to corresponding module handler factories.
      */
     private Map<String, ModuleHandlerFactory> moduleHandlerFactories;
 
     /**
-     * Locker which does not permit rule initialization when the rule engine is stopping.
-     */
-    private boolean isDisposed = false;
```

Comment:
Don't remove this flag. It signals that the dispose operation is started and no more rule initializations are accepted